### PR TITLE
[Tests/CAPI] Use constexpr instead of define and ifdef for tflite @open sesame 04/13 15:19:47

### PR DESCRIPTION
**Pull request by wooksong**
This patch experimentally replaces #define and #ifdef used for checking enable-tensorflow-lite is true with constexpr, which is type-safe replacement for #define.

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

Signed-off-by: Wook Song <wook16.song@samsung.com>

### CODE REVIEW

File: tests/tizen_capi/unittest_tizen_capi.cc [hunk 1]
```diff
@@ -1282,27 +1287,45 @@ TEST (nnstreamer_capi_util, availability_01)
 
   status = ml_check_nnfw_availability (ML_NNFW_TYPE_TENSORFLOW_LITE, ML_NNFW_HW_ANY, &result);
   EXPECT_EQ (status, ML_ERROR_NONE);
-  EXPECT_EQ (result, true);
+  if (is_enabled_tensorflow_lite)
+    EXPECT_EQ (result, true);
+  else
+    EXPECT_NE (result, true);
```
Comment (myungjoo) [inline tests/tizen_capi/unittest_tizen_capi.cc:1293]: How about ```EXPECT_EQ (result, is_enabled_tensorflow_lite)``` ?

File: tests/tizen_capi/unittest_tizen_capi.cc [hunk 2]
```diff
@@ -4526,55 +4673,58 @@ TEST (nnstreamer_capi_singleshot, invoke_dynamic_success_02_p)
...
-  EXPECT_EQ (output_buf2[2], 5.0f);
-  EXPECT_EQ (output_buf2[3], 6.0f);
-  EXPECT_EQ (output_buf2[4], 7.0f);
-  EXPECT_EQ (data_size, 5 * sizeof (float));
+    EXPECT_EQ (output_buf2[0], 3.0f);
+    EXPECT_EQ (output_buf2[1], 4.0f);
+    EXPECT_EQ (output_buf2[2], 5.0f);
+    EXPECT_EQ (output_buf2[3], 6.0f);
+    EXPECT_EQ (output_buf2[4], 7.0f);
+    EXPECT_EQ (data_size, 5 * sizeof (float));
```
Comment (myungjoo) [inline tests/tizen_capi/unittest_tizen_capi.cc:4705]: For floating point values, EXPECT_NEAR/ASSERT_NEAR is more appropriate than EXPECT_EQ/ASSERT_EQ.

File: tests/tizen_capi/unittest_tizen_capi.cc (hunk 1 above)
Comment (wooksong) [reply tests/tizen_capi/unittest_tizen_capi.cc]: @myungjoo That's better. Thanks.

File: tests/tizen_capi/unittest_tizen_capi.cc (hunk 2 above)
Comment (wooksong) [reply tests/tizen_capi/unittest_tizen_capi.cc:4687]: @myungjoo I submit another commit for resolving yout comment. Thanks.

File: tests/tizen_capi/unittest_tizen_capi.cc [hunk 3]
```diff
@@ -4352,7 +4352,7 @@ TEST (nnstreamer_capi_singleshot, invoke_dynamic_success_01_p)
   ml_tensors_data_get_tensor_data (output, 0, (void **) &output_buf,
       &data_size);
 
-  EXPECT_EQ (output_buf[0], 3.0f);
+  EXPECT_NEAR (output_buf[0], 3.0f, 1e-5);
```
Comment (jaeyun-jung) [inline tests/tizen_capi/unittest_tizen_capi.cc]: how about using 'EXPECT_FLOAT_EQ'

Comment (wooksong) [reply tests/tizen_capi/unittest_tizen_capi.cc]: @jaeyun-jung Replce EXPECT_NEAR with EXPECT_FLOAT_EQ.

**Pull request merged**